### PR TITLE
Added variant of JSON benchmarks using Jil

### PR DESCRIPTION
**Pull request by DamianEdwards**
Middleware and MVC versions of JSON scenario using Jil

### CODE REVIEW

File: build/repo.targets [hunk 1]
```diff
@@ -5,8 +5,10 @@
     <CloudScenarios Include="-n Plaintext" />
     <CloudScenarios Include="-n MvcPlaintext" />
 
+    <CloudScenarios Include="-n Jil" />
```
Comment (mikeharder) [inline build/repo.targets]: Move below `Json` for consistency.

File: build/repo.targets [hunk 2]
```diff
@@ -60,12 +62,24 @@
     <PhysicalScenarios Include="-n Json -m https --kestrelTransport Sockets" />
     <PhysicalScenarios Include="-n Json -m https --webHost HttpSys" />
 
+    <PhysicalScenarios Include="-n Jil" />
+
+    <PhysicalScenarios Include="-n Jil --kestrelTransport Sockets" />
```
Comment (mikeharder) [inline build/repo.targets]: Would it be sufficient to just test `Jil` and compare to `Json`?  I don't think the other variants of this scenario are worth the cost.

File: src/Benchmarks/Configuration/Scenarios.cs [hunk 3]
```diff
@@ -41,12 +41,18 @@ public Scenarios(IScenariosConfiguration scenariosConfiguration)
         [ScenarioPath("/json")]
         public bool Json { get; set; }
 
+        [ScenarioPath("/jil")]
```
Comment (mikeharder) [inline src/Benchmarks/Configuration/Scenarios.cs]: New scenarios must also be added to `src/Benchmarks.ServerJob/Scenario.cs` and `src/BenchmarksDriver/Program.cs`.  See https://github.com/aspnet/benchmarks/pull/301/files.

**Comment by mikeharder** [comment]
@DamianEdwards: Should the `Jil` scenarios also be added to the TechEmpower repo?

We will also need to update our PowerBI dashboards to include `Jil` in the `Json` charts.

File: src/Benchmarks/Benchmarks.csproj [hunk 4]
```diff
@@ -17,6 +17,7 @@
 
   <ItemGroup>
     <PackageReference Include="Dapper" Version="$(DapperVersion)" />
+    <PackageReference Include="Jil" Version="2.15.4" />
```
Comment (mikeharder) [inline src/Benchmarks/Benchmarks.csproj]: Move version number to `build/dependencies.props` and reference with a variable.

File: src/Benchmarks/Controllers/HomeController.cs [hunk 5]
```diff
@@ -18,12 +19,19 @@ public IActionResult Plaintext()
...
-        [Produces("application/json")]
+        [Produces("application/json", Type = typeof(JsonOutputFormatter))]
         public object Json()
         {
             return new { message = "Hello, World!" };
         }
-        
+
+        [HttpGet("jil")]
+        [Produces("application/json", Type = typeof(JilOutputFormatter))]
```
Comment (rynowak) [inline src/Benchmarks/Controllers/HomeController.cs]: This doesn't do what you want. The `Type` is there for API Explorer

My suggestion:
1. DI MVC options
2. Get the formatters from options
3. `return new ObjectResult(....) { Formatters { myformatter } };`

File: build/repo.targets (hunk 2 above)
Comment (DamianEdwards) [reply build/repo.targets]: Fair enough, we can always add it back later.

**Comment by DamianEdwards** [comment]
@mikeharder let's run them here first before we decide to add them to official benchmarks.

File: src/Benchmarks/Controllers/HomeController.cs (hunk 5 above)
Comment (DamianEdwards) [reply src/Benchmarks/Controllers/HomeController.cs]: Hmmm, little concerned how much farther away from "idiomatic" this takes us. Is there not a way to do this outside of the controller?

File: src/Benchmarks/Configuration/Scenarios.cs (hunk 3 above)
Comment (DamianEdwards) [reply src/Benchmarks/Configuration/Scenarios.cs]: Done.

File: src/BenchmarksDriver/Program.cs [hunk 6]
```diff
@@ -71,11 +80,13 @@ public class Program
             {
                 { Scenario.Plaintext, _plaintextJob },
                 { Scenario.Json, _jsonJob },
+                { Scenario.Jil, _jilJob },
```
Comment (mikeharder) [inline src/BenchmarksDriver/Program.cs]: If there are no differences between `_jsonJob` and `_jilJob` (I didn't see any), you can delete `_jilJob` and just use `_jsonJob`.

File: build/repo.targets (hunk 2 above)
Comment (mikeharder) [reply build/repo.targets]: @DamianEdwards or @sebastienros: Please remove the extra scenarios before merging.  Let's just run `Jil` and `MvcJil` on Physical.

File: src/Benchmarks/Controllers/HomeController.cs (hunk 5 above)
Comment (rynowak) [reply src/Benchmarks/Controllers/HomeController.cs]: You can use a filter, but you need to do the same dance. The JSON formatter isn't trivially constructable.

It's not really *idomatic* to use two separate JSON formatters in an app. We originally had a feature like this but we cut it for simplification purposes.

Comment (DamianEdwards) [reply src/Benchmarks/Controllers/HomeController.cs]: Fair point, hmmm. Perhaps I'll make it replace the JSON formatter instead.

**Comment by DamianEdwards** [comment]
@sebastienros are we getting this one in soon?

**Pull request merged**